drop-storage/src/lib.rs

```diff
@@ -897,6 +897,37 @@ impl Storage {
         }
     }
 
+    pub fn fetch_temp_locations(&self, transfer_id: Uuid) -> Result<Vec<TempFileLocation>> {
+        let tid = transfer_id.to_string();
+
+        trace!(
+            self.logger,
+            "Fetching temporary file locations";
+            "transfer_id" => &tid
+        );
+
+        let conn = self.pool.get()?;
+
+        let out = conn
+            .prepare(
+                r#"
+                SELECT DISTINCT path_hash, base_dir
+                FROM incoming_paths ip
+                INNER JOIN incoming_path_started_states ipss ON ip.id = ipss.path_id 
+                WHERE transfer_id = ?1
```

Comment:
- DISTINCT usage gives me questions. We might have multiple entries thus it will give us one, but in that case I think LIMIT 1 is a tiny bit more understandable. 

- The code fetches temp file locations for transfers that have a started state but it doesn't care if they have any other state, for example, completed transfers should not have a temporary file left over. TLDR for simplicity - why not return without any join?

Reply:
As per discussion: there can be multiple temp files because download calls might be canceled and another destination used.